Semigroup-Monoid.md

```diff
@@ -0,0 +1,318 @@
+
+
+```kotlin
+fun <A, B> monoidTuple(MA: Monoid<A>, MB: Monoid<B>): Monoid<Tuple2<A, B>> =
+  object: Monoid<Tuple2<A, B>> {
+  //...
+```
+
+^ In this new example, we can see one more way of composing monoids that allow us to operate on more complex types. Using this with a function that produces a tuple, we can define a Monoid for a tuple that will be valid for any tuple where the types it contains also have a Monoid available.
+
+---
+
+# Monoid
+
+Let's have a look to the whole `monoidTuple` function. What we have here is a new `Monoid` in which we are providing more functionality to the `data class` `Tuple2` defining both, the combine extension function and the `empty` method, for a `Tuple2<A, B>`.
+
+```kotlin
+fun <A, B> monoidTuple(MA: Monoid<A>, MB: Monoid<B>): Monoid<Tuple2<A, B>> =
+  object: Monoid<Tuple2<A, B>> {
+
+    override fun Tuple2<A, B>.combine(y: Tuple2<A, B>): Tuple2<A, B> {
+      val (xa, xb) = this
+      val (ya, yb) = y
+      return Tuple2(MA.run { xa.combine(ya) }, MB.run { xb.combine(yb) })
+    }
+
+    override fun empty(): Tuple2<A, B> = Tuple2(MA.empty(), MB.empty())
+  }
+```  
+
+^ Let's have a deeper look to the monoidTuple function. This is a new Monoid in which we are providing more functionality to the data class Tuple2 defining both, the combine extension function and the empty method, for a Tuple2 of A and B.
+^ The implementation of the combine method provides that behaviour to the monoids that conform the Tuple2, and the empty method is overrided here adding that functionallity to the monoids in the Tuple2 of A and B.
+^ We can see an example of how this function can be used, in the next slide.
+
+---
+
+# Monoid
+
+In this example, we see a function where the `foldMap` receives a Monoid `M` and a function that takes an element from the list and returns the type of the `Monoid`.
+The `foldMap` usses the `Monoid.empty` as base instance and uses the function `Tuple2.combine` for combining the base instance with the result of the fuction doing it recursively among every element in the list.
+This way we are able to combine both values in one pass!
+
+```kotlin
+with(ListK.foldable()) {
+  val m = monoidTuple(Int.monoid(), String.monoid())
+  val list = listOf(1, 1).k()
+
+  list.foldMap(m) { n: Int ->
+   Tuple2(n, n.toString())
+  }
+}
+// Tuple2(a=2, b=11)
+```
+
+^ In this example, we see a function where the foldMap receives a Monoid M and a function that takes an element from the list and returns the type of the Monoid. In this case is a Tuple composed by a Monoid of Int and a Monoid of String.
+^ The foldMap uses the Monoid.empty as base instance and uses the function Tuple2.combine for combining the base instance with the result of the fuction doing it recursiverly among every element in the list.
+^ Magically, we are able to combine both values in one pass!
+
+---
+
+Thanks for watching!
```

Comment:
Missing closing sentence where we recap what we've learned as shown in other videos